frontend/javascripts/admin/user/permissions_and_teams_modal_view.js

```diff
@@ -0,0 +1,350 @@
+          }
+        }
+
+        // server-side validation can reject a user's new teams
+        return updateUser(newUser).then(
+          serverUser => Promise.resolve(serverUser),
+          () => Promise.reject(user),
+        );
+      }
+      return Promise.resolve(user);
+    });
+
+    Promise.all(newUserPromises).then(
+      newUsers => {
+        this.props.onChange(newUsers);
+      },
+      () => {
+        // do nothing and keep modal open
+      },
+    );
+  };
+
+  handlePermissionChanged = (evt: SyntheticInputEvent<>) => {
+    const selectedPermission: $Values<typeof PERMISSIONS> = evt.target.value;
+    this.setState({ selectedPermission });
+  };
+
+  handleSelectTeamRole(teamName: string, isTeamManager: boolean) {
+    const team = this.state.teams.find(t => t.name === teamName);
+
+    if (team) {
+      const selectedTeam = { id: team.id, name: teamName, isTeamManager };
+      this.setState(prevState => ({
+        selectedTeams: update(prevState.selectedTeams, {
+          [teamName]: { $set: selectedTeam },
+        }),
+      }));
+    }
+  }
+
+  handleUnselectTeam(teamName: string) {
+    this.setState(prevState => ({
+      selectedTeams: update(prevState.selectedTeams, {
+        $unset: [teamName],
+      }),
+    }));
+  }
+
+  getTeamComponent(team: APITeam, isDisabled: boolean) {
+    return (
+      <Checkbox
+        value={team.name}
+        checked={_.has(this.state.selectedTeams, team.name)}
+        disabled={isDisabled}
+        onChange={(event: SyntheticInputEvent<>) => {
+          if (event.target.checked) {
+            this.handleSelectTeamRole(team.name, false);
+          } else {
+            this.handleUnselectTeam(team.name);
+          }
+        }}
+      >
+        {team.name}
+      </Checkbox>
+    );
+  }
+
+  getRoleComponent(team: APITeam, isDisabled: boolean) {
+    const selectedTeam = this.state.selectedTeams[team.name];
+    let selectedValue = null;
+    if (selectedTeam) {
+      selectedValue = selectedTeam.isTeamManager ? ROLES.teammanager : ROLES.user;
+    }
+
+    return (
+      <RadioGroup
+        size="small"
+        style={{ width: "100%", paddingBottom: 8 }}
+        value={selectedValue}
+        disabled={!_.has(this.state.selectedTeams, team.name) || isDisabled}
+        onChange={({ target: { value } }) =>
+          this.handleSelectTeamRole(team.name, value === ROLES.teammanager)
+        }
+      >
+        <RadioButton value={ROLES.teammanager}>Team Manager</RadioButton>
+        <RadioButton value={ROLES.user}>Member</RadioButton>
+      </RadioGroup>
+    );
+  }
+
+  getPermissionSelection(onlyEditingSingleUser: boolean, isUserAdmin: boolean) {
+    const roleStyle = { fontWeight: "bold" };
+    const explanationStyle = { color: "rgba(0, 0, 0, 0.55", paddingBottom: 12 };
+    return (
+      <React.Fragment>
+        <h4>
+          Organization Permissions{" "}
+          <a href="https://docs.webknossos.org/guides/users">
```

Comment:
@MichaelBuessemeyer Could you make this open in a new tab, that's what we usually do for documentation links :)